quickwit/quickwit-metastore/src/metastore/file_backed/mod.rs

```diff
@@ -505,13 +506,26 @@ impl MetastoreService for FileBackedMetastore {
         let retention_policy_opt = request.deserialize_retention_policy()?;
         let search_settings = request.deserialize_search_settings()?;
         let indexing_settings = request.deserialize_indexing_settings()?;
+        let doc_mapping = request.deserialize_doc_mapping()?;
         let index_uid = request.index_uid();
 
+        // verify the new mapping is coherent
+        let doc_mapper_builder = DefaultDocMapperBuilder {
+            doc_mapping: doc_mapping.clone(),
+            default_search_fields: search_settings.default_search_fields.clone(),
+        };
+        doc_mapper_builder
+            .try_build()
+            .map_err(|e| MetastoreError::InvalidArgument {
+                message: format!("invalid mapping update: {e}"),
+            })?;
+
```

Comment:
I think it's risky to have this validation duplicated here and in the postgres implem. We could:
- group `IndexMetadata.set_search_settings/set_indexing_settings/set_retentionpolicy/set_doc_mapping` into a single `IndexMetadata.set_index_config(all 4 settings objects)` and perform this validation there
- perform the validations on UpdateIndexRequest's deserialization